[Term Entry] JavaScript DOM Manipulation: .querySelectorAll()

DimitrovErik opened the pull request:
<!--
👋 Hi, thanks for submitting a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->

- Adding folders and a file under :
content/javascript/concepts/dom-manipulation/terms/queryselectorall/queryselectorall.md
- Creating a new entry file on the .querySelectorAll() term under the DOM concept in JavaScript
- Closes issue #6239 

### Issue Solved

<!--
Please use the following format(s) to link the issue numbers that your contribution addresses:
Closes #issue-number (where `issue-number` corresponds to the PRs' respective issue)
-->

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/docs?pullRequest=6296) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=6296) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=6296) it.</sub>

mamtawardhani commented:
> [![CLA assistant check](https://camo.githubusercontent.com/bb14f9741a98be03eb447ca21a4a5e96201df401e5be70f9ef52130f4e1ff903/68747470733a2f2f636c612d617373697374616e742e696f2f70756c6c2f62616467652f6e6f745f7369676e6564)](https://cla-assistant.io/Codecademy/docs?pullRequest=6296) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=6296) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=6296) it.

please sign the CLA @DimitrovErik

mamtawardhani commented on content/javascript/concepts/dom-manipulation/terms/queryselectorall/queryselectorall.md:
```diff
@@ -0,0 +1,90 @@
+
+# Document: querySelectorAll() method
+
+The [Document](https://developer.mozilla.org/en-US/docs/Web/API/Document) method **querySelectorAll()** returns a static (not live) NodeList representing a list of the document's elements that match the specified group of selectors.
```
```suggestion
The document method **`querySelectorAll()`** returns a static (not live) `NodeList` of all elements that match the specified group of selectors.
```

mamtawardhani commented on content/javascript/concepts/dom-manipulation/terms/queryselectorall/queryselectorall.md:
````diff
@@ -0,0 +1,90 @@
+```js
+querySelectorAll(selectors);
+```
+
````
explain what is the selector parameter here in the following format:

- `selectors`: <description>

mamtawardhani commented on content/javascript/concepts/dom-manipulation/terms/queryselectorall/queryselectorall.md:
```diff
@@ -0,0 +1,90 @@
+
+## Example
+
+To obtain a [NodeList](https://developer.mozilla.org/en-US/docs/Web/API/NodeList) of all of the [< p >](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/p) elements in the document:
```
```suggestion
In this example, a `NodeList` of all `<p>` elements in the document is obtained:
```

mamtawardhani commented on content/javascript/concepts/dom-manipulation/terms/queryselectorall/queryselectorall.md:
````diff
@@ -0,0 +1,90 @@
+const matches = document.querySelectorAll("p");
+```
+
+This example returns a list of all [< div >](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/div) elements within the document with a class of either note or alert:
````
```suggestion
This example returns a list of all `<div>` elements in the document with a class of either `note` or `alert`:
```

mamtawardhani commented on content/javascript/concepts/dom-manipulation/terms/queryselectorall/queryselectorall.md:
````diff
@@ -0,0 +1,90 @@
+const matches = document.querySelectorAll("div.note, div.alert");
+```
+
+Here, we get a list of < p > elements whose immediate parent element is a < div > with the class highlighted and which are located inside a container whose ID is test.
````
```suggestion
In this example, we get a list of `<p>` elements whose immediate parent is a `<div>` with the class `highlighted`, and which are inside a container with the ID `test`:
```

mamtawardhani commented on content/javascript/concepts/dom-manipulation/terms/queryselectorall/queryselectorall.md:
````diff
@@ -0,0 +1,90 @@
+We can currently support:
+
+- JavaScript
+
+We need to use different libraries or tools to parse and manipulate HTML content in ways similar to JavaScript's **.querySelectorAll()**:
+- Python
+- Ruby
+- C++
+- C#
+- Go
+- PHP
+
+See [content-standards.md](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) for more details!
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>querySelectorAll Example</title>
+</head>
+<body>
+  <p>This is paragraph 1.</p>
+  <p>This is paragraph 2.</p>
+  <p>This is paragraph 3.</p>
+
+  <script>
+    // Select all <p> elements on the page
+    const paragraphs = document.querySelectorAll('p');
+
+    // Loop through each <p> element and add some extra text
+    paragraphs.forEach((para, index) => {
+      para.textContent += ` - Updated by querySelectorAll (${index + 1})`;
+    });
+  </script>
+</body>
+</html>
+
+```
````
```suggestion
```

mamtawardhani replied:
we dont need a codebyte for this PR

DimitrovErik commented:
Hey @mamtawardhani 
I’ve addressed all the feedback. Let me know if further adjustments are required.
Thank you for your detailed commit suggestions 🙌🏼 – I really appreciate the guidance, and I’ve learned a lot from your feedback!

mamtawardhani reviewed (approved):
Thank you for contributing to Codecademy Docs @DimitrovErik  😄 
The entry looks good for a second round of review! 🚀

DimitrovErik commented:
@mamtawardhani I just fix the formatting issue:
I ran Prettier to automatically fix formatting issues that were causing the format:verify check to fail. 
The changes were committed, and the file is now properly formatted according to our Prettier configuration.
The following command was used to resolve the issue:
npx prettier --write <file-path>
Everything should be good now, and the formatting has been verified🙌🏼🙃

Sriparno08 reviewed (approved):
Looks good, @DimitrovErik!
